### PR TITLE
Drop node 0.10/0.12 support and update dependencies

### PR DESCRIPTION
**Pull request by gigabo**
Had to clean up some indentation due to stricter new eslint.

Had to port inquirer usage to new promise-based interface.

Had to _add_ a dep on `eslint-plugin-flowtype` since `eslint-config-babel` now
requires it as a peerDependency.  Removed `eslint-plugin-flow-vars` as the
other side of that change.

Didn't do asini-changelog deps here.  Will do those separately.

### CODE REVIEW

**Comment by gigabo** [comment]
Heh, the newest eslint doesn't even support node 0.12.

```
> asini@1.3.0 lint /home/travis/build/asini/asini
> eslint src test
/home/travis/build/asini/asini/node_modules/eslint/bin/eslint.js:16
const useStdIn = (process.argv.indexOf("--stdin") > -1),
^^^^^
SyntaxError: Use of const in strict mode.
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```

**Comment by evocateur** [comment]
hah, I have these exact changes locally. I didn't realize eslint 3.x had dropped 0.12 support, and seems like a strong argument for asini dropping 0.12 as well (at least explicitly?).

File: package.json [hunk 1]
```diff
@@ -54,18 +54,19 @@
...
     "babel-preset-es2015": "^6.6.0",
     "babel-register": "^6.7.2",
     "cross-env": "^1.0.8",
-    "eslint": "^2.3.0",
-    "eslint-config-babel": "^1.0.1",
-    "eslint-plugin-babel": "^3.3.0",
+    "eslint": "^3.10.2",
+    "eslint-config-babel": "^3.0.0",
+    "eslint-plugin-babel": "^4.0.0",
     "eslint-plugin-flow-vars": "^0.5.0",
```
Comment (evocateur) [inline package.json]: We can remove `eslint-plugin-flow-vars`, too, as it is no longer a peer dependency of `eslint-config-babel`.

**Review by evocateur — APPROVED** [review]
If we drop node 0.12, which I am in favor of, I agree we should bump the major. We might consider adding any other breaking changes (if any?) before cutting a 2.0.

File: package.json (hunk 1 above)
Comment (gigabo) [reply package.json]: Oh, nice catch!

**Comment by gigabo** [comment]
Okay, rather than backing out the `eslint` upgrade I've actually gone the other direction and upgraded _everything_ here.  This required dropping _both_ node 0.10 _and_ 0.12 support, so this would close #61 and then some.  Looks like 0.12 is [end-of-life at the end of the year](https://github.com/nodejs/LTS).

This is a breaking change for anybody still using 0.10 or 0.12, so we'd do a major version bump.

I don't think we have any other breaking changes close enough to wait for.

@evocateur, @rygine anything spring to mind?

**Comment by evocateur** [comment]
@gigabo Maybe extricate `asini-changelog`'s circular-ish dependency on `asini` as part of 2.0?

Otherwise, LGTM

**Comment by gigabo** [comment]
Oh yeah, good call.  I'll put up a PR for that once this lands (since they'll probably conflict).

**Pull request merged**